internal/pkg/aws/ecr/ecr.go

```diff
@@ -97,22 +99,34 @@ func (i Image) imageIdentifier() *ecr.ImageIdentifier {
 // ListImages calls the ECR DescribeImages API and returns a list of
 // Image metadata for images in the input ECR repository name.
 func (s Service) ListImages(repoName string) ([]Image, error) {
-	// TODO: handle paginated responses
-	resp, err := s.ecr.DescribeImages(&ecr.DescribeImagesInput{
-		RepositoryName: aws.String(repoName),
-	})
-
-	if err != nil {
-		return nil, fmt.Errorf("ecr repo %s describe images: %w", repoName, err)
-	}
-
 	var images []Image
-	for _, imageDetails := range resp.ImageDetails {
-		images = append(images, Image{
-			Digest: *imageDetails.ImageDigest,
-		})
+	nextTokenExist := false
+	var resp *ecr.DescribeImagesOutput
+	var err error
+	for {
```

Comment:
I think you could simplify a lot of this logic by making an initial call to `DescribeImages` and then loop while `resp.NextToken != nil` instead of trying to handle both scenarios (initial call and follow-up calls) within the same infinite loop. If that doesn't make sense let me know and I can help provide an example.

Reply:
Addressed